src/Algebra/Graph/AdjacencyIntMap/Algorithm.hs

```diff
@@ -115,6 +118,56 @@ dfs vs = concatMap flatten . dfsForestFrom vs
 reachable :: Int -> AdjacencyIntMap -> [Int]
 reachable x = dfs [x]
 
+-- | Compute the forest of a graph's vertices in breadth first order.
+-- Complexity: /O(n+m*log n)/ time and /O(n+m)/ space.
+--
+-- @
+-- bfsForest 'empty'                         == []
+-- 'forest' (bfsForest $ 'edge' 1 1)           == 'vertex' 1
+-- 'forest' (bfsForest $ 'edge' 1 2)           == 'edge' 1 2
+-- 'forest' (bfsForest $ 'edge' 2 1)           == 'vertices' [1,2]
+-- 'isSubgraphOf' ('forest' $ bfsForest x) x   == True
+-- bfsForest . 'forest' . bfsForest          == bfsForest
+-- 'forest' (bfsForest ('circuit' [1..5] + 'transpose' ('circuit' [1..5]))) == 'path' [1,2,3] + 'path' [1,5,4]
+-- @
+bfsForest :: AdjacencyIntMap -> Forest Int
+bfsForest g = bfsForestFrom' (vertexList g) g
+
+-- | Like 'bfsForest', but the traversal is seeded by a list of
+-- vertices.  Let /L/ be the number of seed vertices. Complexity:
+-- /O(n+(L+m)*log n)/ time and /O(n+m)/ space.
+--
+-- @
+-- 'forest' (bfsForestFrom [1,2] $ 'edge' 1 2) == 'vertices' [1,2]
+-- 'forest' (bfsForestFrom [2]   $ 'edge' 2 1) == 'vertex' 2
+-- 'forest' (bfsForestFrom [3]   $ 'edge' 1 2) == empty
+-- 'forest' (bfsForestFrom [3] ('circuit' [1..5] + 'transpose' ('circuit' [1..5]))) == 'path' [3,2,1] + 'path' [3,4,5]
+-- @
+bfsForestFrom :: [Int] -> AdjacencyIntMap -> Forest Int
+bfsForestFrom vs g = bfsForestFrom' [ v | v <- vs, hasVertex v g] g
+
+bfsForestFrom' :: [Int] -> AdjacencyIntMap -> Forest Int
+bfsForestFrom' vs g = evalState (bff vs) IntSet.empty where
+  bff [] = return []
+  bff (v:vs) = discovered v >>= \case
+    False -> bff vs
+    True -> (:) <$> unfoldTreeM_BF walk v <*> bff vs
+  walk v = (v,) <$> adjacentM v
+  adjacentM v = filterM discovered $ IntSet.toList (postIntSet v g)
+  discovered v = do new <- gets (not . IntSet.member v)
+                    when new $ modify' (IntSet.insert v)
+                    return new
+
+-- | Like 'bfsForestFrom' with the resulting forest flattened to a
+-- list of vertices. Complexity: /O(n+(L+m)*log n)/ time and /O(n+m)/
```

Comment:
`L` is undefined.

Reply:
It's mentioned in the documentation for `bfsForestFrom` as the number of seed vertices. I should probably restate it here?

 I also wasn't sure what to call it, so I went with L... perhaps something else would be clearer?

Reply:
Yes, please redefine `L`. Ideally, each function should have a self-contained description.

Reply:
I switched to `s`, which I think makes more sense. Also restated it in both places it was missing

Reply:
Actually, `s` is a bit awkward, since in other modules `s` stands for the `size` of algebraic expressions. Let's keep `L` -- in fact, I already use `L` to denote the length of the input list in some complexity bounds.